### PR TITLE
feat: workflow library followups

### PR DESCRIPTION
**Pull request by psychedelicious**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [x] Bug Fix
- [x] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Description

Workflow library followup changes, based on #5148.

- fix edge case in image details viewer for workflows and metadata
- fix issue with menus not closing when clicking reactflow
- fix bug where "(copy)" was being appended to workflow name on save
- improve UI/UX for save/as workflows, use persistent updated toast
- use menu instead of big row of buttons for workflow actions
- fix pydantic invocation type mismatch (paste & center pad crop had gotten mixed up)
- use templates for edge validation (addresses obscure edge case)
- improve pydantic workflow schemas handling of id
- track & indicate workflow saved status
- disallow loading/deleting of opened workflow

## Related Tickets & Documents

- Related #5148

## QA Instructions, Screenshots, Recordings

https://github.com/invoke-ai/InvokeAI/assets/4822129/ff604c62-1bfa-4fd9-9651-4d73958ad906

## Added/updated tests?

- [ ] Yes
- [ ] No : _please replace this line with details on why tests
      have not been included_

### CODE REVIEW

**Comment by hipsterusername** [comment]
@psychedelicious is this supposed to be merging into main?

**Comment by psychedelicious** [comment]
@hipsterusername 
it's based on #5148 which doesn't have approval yet

File: invokeai/app/invocations/image.py
Comment (psychedelicious) [inline invokeai/app/invocations/image.py:1]: This fixes a bad merge that resulted in the two invocations having their decorators swapped.

File: invokeai/app/services/workflow_records/workflow_records_common.py
Comment (psychedelicious) [inline invokeai/app/services/workflow_records/workflow_records_common.py:1]: The change here (and in the sqlite impl) is to not add the id field to workflows that don't have one. Instead, the two models are strictly defined:
- `WorkflowWithoutID`: used when creating a new workflow - backend creates IDs, so the client should never provide a workflow with an ID when creating
- `Workflow`: used when modifying an existing workflow

File: invokeai/frontend/web/src/features/gallery/components/ImageMetadataViewer/ImageMetadataViewer.tsx
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/gallery/components/ImageMetadataViewer/ImageMetadataViewer.tsx:1]: - Making the tabs lazy reduces network requests when skipping through images
- Making the tabs never-disabled prevents an awkward UX where you click on a workflow, nav to a different image that doesn't have one, and end up on a disabled tab

File: invokeai/frontend/web/src/features/nodes/components/NodeEditor.tsx
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/nodes/components/NodeEditor.tsx:1]: The workflow buttons have been organized. The reset of the changes in `features/nodes/components/` and `features/workflowLibrary/components|hooks` support this.

File: invokeai/frontend/web/src/common/hooks/useGlobalMenuCloseTrigger.ts
Comment (psychedelicious) [inline invokeai/frontend/web/src/common/hooks/useGlobalMenuCloseTrigger.ts:1]: The logic in this hook was present in a couple places. It has been put into this module and its state renamed to better represent its purpose.

File: invokeai/frontend/web/src/theme/theme.ts
Comment (psychedelicious) [inline invokeai/frontend/web/src/theme/theme.ts:1]: Some minor theme changes.

**Review by hipsterusername — APPROVED** [review]

**Pull request merged**